### PR TITLE
[SG-742]

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

### Two Step Login Cleanup
- Remove `alwaysRemember` code from base Login Component
- Change flow to always land on email screen even if an email is remembered
- Add Remember Email functionality back to browser, now that it is feasible with all changes.

### Libs
- **libs/angular/src/components/login.component.ts:** Remove `alwaysRememberEmail` functionality and landing on the password screen if an email is remembered.

### Web
- **Web State Service** Remove get and set `rememberEmail` state variable. We no longer need this with the presence of the LoginService. We can infer this value on initial load by the presence of a rememberedEmail.

### Browser
- **HomeComponent:** Add remember email option
- **LoginComponent:** Remove unused variable

## Screenshots

https://user-images.githubusercontent.com/8926729/205959961-28442966-4515-4f78-89c6-04e0d44d44b5.mov

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/browser/src/popup/accounts/home.component.html [hunk 1]
```diff
@@ -22,10 +31,10 @@
     </form>
     <p class="createAccountLink">
       {{ "newAroundHere" | i18n }}
-      <a routerLink="/register">{{ "createAccount" | i18n }}</a>
+      <a routerLink="/register" (click)="setFormValues()">{{ "createAccount" | i18n }}</a>
```
Comment (JaredSnider-Bitwarden) [inline apps/browser/src/popup/accounts/home.component.html:34]: I had to pull this code down and test this myself to see if there actually was a race condition here or not. It wasn't clear from researching online what the order of execution would be in this case b/w routerlink and the (click), but I think, based on my tests, that it is fine.  The browser consistently and fully ran the click function before navigating away.

File: apps/browser/src/popup/scss/pages.scss [hunk 2]
```diff
@@ -143,6 +143,12 @@ body.body-full {
   padding: 30px 10px 0 10px;
 }
 
+.remember-email-check {
+  padding-top: 8px;
+  padding-left: 10px;
+  padding-bottom: 18px;
+}
```
Comment (JaredSnider-Bitwarden) [inline apps/browser/src/popup/scss/pages.scss:150]: Why the choice to use custom CSS class vs TailwindCss padding classes?  Are we not using Tailwind in the browser extension yet?

File: apps/web/src/app/core/state/state.service.ts [hunk 3]
```diff
@@ -48,23 +48,6 @@ export class StateService extends BaseStateService<GlobalState, Account> {
     await super.addAccount(account);
   }
 
-  async getRememberEmail(options?: StorageOptions) {
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/core/state/state.service.ts:51]: I don't yet fully understand all the places where the different state services are used, but I was curious why these methods were removed from this state service but not from the others?  For example, the `libs/angular/src/components/login.component.ts` on line 101 still calls `await this.stateService.getRememberedEmail()`.

Comment (differsthecat) [reply apps/web/src/app/core/state/state.service.ts:51]: There's a very slight difference if you look closely; web had/has an additional state value that's a bool: `rememberEmail`, not `rememberedEmail`. This was to capture the remember email checkbox essentially. This value can be inferred by the presence of a value in `rememberedEmail`, so I removed it as it is no longer needed.

File: apps/browser/src/popup/accounts/home.component.html (hunk 1 above)
Comment (differsthecat) [reply apps/browser/src/popup/accounts/home.component.html:34]: I can change the click handler to include the routing, but there are styles that are hinging on that routerLink being present. If you think it makes more sense, I can move the routing to a function instead, and stop the propagation of the routerLink event.

File: apps/web/src/app/core/state/state.service.ts (hunk 3 above)
Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/core/state/state.service.ts:51]: Ah, thank you for the clarification, and sorry for missing that difference.

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
Looks great!

**Comment by differsthecat** [comment]
QA approved, so removing `needs-qa` label

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Pull request merged**